gtars/src/common/models/region_set.rs

```diff
@@ -25,14 +44,95 @@ impl TryFrom<&Path> for RegionSet {
     /// # Arguments:
     /// - value: path to bed file on disk.
     fn try_from(value: &Path) -> Result<Self> {
-        let regions = extract_regions_from_bed_file(value)?;
-        Ok(RegionSet { regions })
+        let path = value;
+        println!(
+            "Initiating regionSet object from: {}",
+            value.to_str().unwrap()
+        );
+
+        let mut new_regions: Vec<Region> = Vec::new();
+
+        let reader = match path.is_file() {
+            true => get_dynamic_reader(path).expect("!Can't read file"),
+            false => get_dynamic_reader_from_url(path).expect("!Can't read file or url!"),
+        };
+
+        let mut header: String = String::new();
+
+        for line in reader.lines() {
+            let string_line = line?;
+
+            let parts: Vec<String> = string_line.split('\t').map(|s| s.to_string()).collect();
+
+            if parts.len() < 3 {
+                if string_line.starts_with("browser")
+                    | string_line.starts_with("track")
+                    | string_line.starts_with("#")
+                {
+                    header.push_str(&string_line);
+                }
+                continue;
+            }
+
+            new_regions.push(Region {
+                chr: parts[0].clone(),
```

Comment:
Do you need to clone here? Can the `Region` struct not just take ownership?

Reply:
It is necessary, i have determined

Reply:
`.to_owned()` and `.to_string()` work as well, so maybe we can use one of those. My point is that the `Region` struct should be able to take ownership of this data... so doing so without cloning feels like a good idea.